vault/resource_consul_secret_backend_role.go

```diff
@@ -125,15 +155,22 @@ func consulSecretBackendRoleWrite(d *schema.ResourceData, meta interface{}) erro
 	path := consulSecretBackendRolePath(backend, name)
 
 	policies := d.Get("policies").([]interface{})
+	consulPolicies := d.Get("consul_policies").([]interface{})
 	roles := d.Get("consul_roles").(*schema.Set).List()
+	serviceIdentities := d.Get("service_identities").(*schema.Set).List()
+	nodeIdentities := d.Get("node_identities").(*schema.Set).List()
 
-	if len(policies) == 0 && len(roles) == 0 {
-		return fmt.Errorf("policies or consul_roles must be set")
+	if len(consulPolicies) == 0 && len(roles) == 0 &&
```

Comment:
I think you can use [`AtLeastOneOf`](https://github.com/hashicorp/terraform-plugin-sdk/blob/main/helper/schema/schema.go#L286-L298) to have Terraform perform this validation before an apply? Might be worth a try.

Reply:
@benashz - Do you happen to know when one would reach for `AtLeastOneOf` if not in this case? It seems like a good thing to catch known configuration issues before an apply and before sending a request to Vault (since we know it fill fail). I agree though -- probably use `AtLeastOneOf` or let Vault take care of validation.

Reply:
Yes, Vault returns an error when none of the above are provided so I think it could be changed to rely on Vault's handling of this situation instead.

Reply:
I opted to fully remove this check instead of using `AtLeastOneOf` because this is logic specifically handled by Vault, which will return a very specific error about missing all of the parameters. Checking in the provider is essentially duplicate code.
https://github.com/hashicorp/terraform-provider-vault/pull/1518/commits/0283556698a7355551d8ee4a2637d696ffc9eb55